core/drlibc/drlibc.c

```diff
@@ -188,6 +190,40 @@ get_cache_line_size(OUT size_t *dcache_line_size, OUT size_t *icache_line_size)
 #    endif
     return false;
 }
+
+/* Reads the block size in bytes required by cache instructions like the DC ZVA
+ * (Data Cache Zero by Address) instruction.
+ * Returns false if no value was written.
+ * This is required to be called at init time when linked into DR. This is to
+ * avoid races and write issues with the static variable used.
+ */
+bool
+get_dcache_zero_blk_size(OUT size_t *dzcva_block_size)
+{
+#    ifndef DR_HOST_NOT_TARGET
+    static size_t dcache_zero_info = 0;
+
+    /* "Data Cache Zero ID register" contains:
+     * DCZID_EL0 [3:0]   : Log2 of number of words (4 bytes) of the block size.
+     *                     The maximum size supported is 2KB => value = 9, i.e.
+                           2^9 x 4 = 512 x 4 = 2048 bytes.
+     * https://developer.arm.com/documentation/ddi0595/2021-09/AArch64-Registers/
+     * DCZID-EL0--Data-Cache-Zero-ID-register
+     */
+    if (dcache_zero_info == 0)
+        __asm__ __volatile__("mrs %0, dczid_el0" : "=r"(dcache_zero_info));
+    if (dzcva_block_size != NULL)
+        *dzcva_block_size = 4 << (dcache_zero_info & 0xf);
+    return true;
+#    endif
+    return false;
+}
+
```

Comment:
Added `set_dcache_zva_size()`.                                            
To be clear: we should set the default in the case of `DR_HOST_NOT_TARGET` but it can be overriden by `set_dcache_zva_size()`?

Reply:
> To be clear: we should set the default in the case of DR_HOST_NOT_TARGET but it can be overriden by set_dcache_zva_size()?

Yes.

Changing dcache_zva_size would change how some instrs are encoded and decoded. I wonder if that can cause any issues due to OPSZ_CACHE_LINE changing. I guess we can verify that when we use `set_dcache_zva_size`/`drutil_insert_get_mem_addr ` in a test, in this or a later PR.

Reply:
Xref https://github.com/DynamoRIO/dynamorio/pull/5776#discussion_r1043395057 where we have another "global" ISA state.  Maybe instead of completely separate set/get routine for each of these we should combine them into a config struct to make it easier to add more; so we'd have something like dr_set_isa_config(dr_isa_config_t *config) or sthg with a size field so we can append.